test_case_gen.py

```diff
@@ -0,0 +1,120 @@
+"""
+Generates Test Cases 0, 1 and saves them to json
+"""
+from network_simulator.network_topology import *
+from network_simulator.host import *
+from network_simulator.common import *
+from network_simulator.router import *
+from network_simulator.flow import *
+from network_simulator.link import *
+
+if __name__ == "__main__":
+    # test case 0
+    links = list()
+    l1 = Link("H1", "H2", h1, h2, LinkBuffer(64 * KILOBYTE), 0.01,
+              10 * KILOBYTE, False)
+    links.append(l1)
+    h1.link = l1
+    h2.link = l1
+
+    hosts = list()
+    h1 = Host("H1")
+    h2 = Host("H2")
+    hosts.append(h1)
+    hosts.append(h2)
+
+    # generate for FlowDummy
+    flows = list()
+    f1 = FlowDummy("F1", "H1", "H2", h1, h2, INITIAL_WINDOW_SIZE_PACKETS,
+                                  set(), list(), 20 * MEGABYTE, 1)
+    flows.append(f1)
+    h1.flows["F1"] = f1
+
+    # (links, flows, hosts, routers) but we have no routers
+    network = NetworkTopology(links, flows, hosts, None)
+    network.write_to_json("test_case0_dummy.json")
+
+    # generate for FlowReno
+    flows = list()
+    f1 = FlowReno("F1", "H1", "H2", h1, h2, INITIAL_WINDOW_SIZE_PACKETS,
+                                  set(), list(), 20 * MEGABYTE, 1)
+    flows.append(f1)
+    h1.flows["F1"] = f1  # overwrites existing FlowDummy in h1
+    network = NetworkTopology(links, flows, hosts, None)
+    network.write_to_json("test_case0_reno.json")
+
+    # generate for FlowFast
+    flows = list()
+    f1 = FlowFast("F1", "H1", "H2", h1, h2, INITIAL_WINDOW_SIZE_PACKETS,

```

Comment:
Same comments as before. These apply to the rest of the file too so I will not repeat them.